File: Tasks/Replace.cpp
#include <iostream>
#include <string>
using namespace std;
const string replace(string str = "", const string substr = "", const string newstr = "")
{
  if (substr == "") return str;
  const int positionofsubstr = str.find(substr);

  return str.erase(positionofsubstr, substr.size()).insert(positionofsubstr, newstr);
}
int main()
{
  cout << replace("Hello <username> and bye!", "<username>", "Marcus");
  return 0;
}

File: Tests/replace.js
'use strict';

const test = require('../test.js');

const cases = [
  [
    ['Hello <username> and bye!', '<username>', 'Marcus'],
    'Hello Marcus and bye!',
  ],
  [['Hello X and bye!', 'X', 'Marcus'], 'Hello Marcus and bye!'],
  [['X and bye!', 'X', 'Marcus'], 'Marcus and bye!'],
  [['Hello X', 'X', 'Marcus'], 'Hello Marcus'],
  [['Hello X', 'Y', 'Marcus'], 'Hello X'],
  [['Hello X', '', 'Marcus'], 'Hello X'],
  [['', 'Y', 'Marcus'], ''],
];

module.exports = test(cases);
